manifest/provider/import.go

```diff
@@ -0,0 +1,192 @@
+			Detail:   err.Error(),
+		})
+		return resp, nil
+	}
+	rcl := client.Resource(gvr)
+
+	var ro *unstructured.Unstructured
+	if ns {
+		ro, err = rcl.Namespace(namespace).Get(ctx, name, metav1.GetOptions{})
+	} else {
+		ro, err = rcl.Get(ctx, name, metav1.GetOptions{})
+	}
+	if err != nil {
+		resp.Diagnostics = append(resp.Diagnostics, &tfprotov5.Diagnostic{
+			Severity: tfprotov5.DiagnosticSeverityError,
+			Summary:  fmt.Sprintf("Failed to get resource %+v from API", io),
+			Detail:   err.Error(),
+		})
+		return resp, nil
+	}
+	s.logger.Trace("[ImportResourceState]", "[API Resource]", ro)
+
+	objectType, err := s.TFTypeFromOpenAPI(ctx, gvk, false)
+	if err != nil {
+		resp.Diagnostics = append(resp.Diagnostics, &tfprotov5.Diagnostic{
+			Severity: tfprotov5.DiagnosticSeverityError,
+			Summary:  fmt.Sprintf("Failed to determine resource type from GVK: %s", gvk),
+			Detail:   err.Error(),
+		})
+		return resp, nil
+	}
+
+	fo := RemoveServerSideFields(ro.UnstructuredContent())
+	nobj, err := payload.ToTFValue(fo, objectType, tftypes.NewAttributePath())
+	if err != nil {
+		resp.Diagnostics = append(resp.Diagnostics, &tfprotov5.Diagnostic{
+			Severity: tfprotov5.DiagnosticSeverityError,
+			Summary:  "Failed to convert unstructured to tftypes.Value",
+			Detail:   err.Error(),
+		})
+		return resp, nil
+	}
+	nobj, err = morph.DeepUnknown(objectType, nobj, tftypes.NewAttributePath())
+	if err != nil {
+		resp.Diagnostics = append(resp.Diagnostics, &tfprotov5.Diagnostic{
+			Severity: tfprotov5.DiagnosticSeverityError,
+			Summary:  "Failed to backfill unknown values during import",
+			Detail:   err.Error(),
+		})
+		return resp, nil
+	}
+	s.logger.Trace("[ImportResourceState]", "[tftypes.Value]", nobj)
+
+	newState := make(map[string]tftypes.Value)
+	wftype := rt.(tftypes.Object).AttributeTypes["wait_for"]
+	newState["manifest"] = tftypes.NewValue(tftypes.Object{AttributeTypes: map[string]tftypes.Type{}}, nil)
+	newState["object"] = morph.UnknownToNull(nobj)
+	newState["wait_for"] = tftypes.NewValue(wftype, nil)
+	nsVal := tftypes.NewValue(rt, newState)
+
+	impState, err := tfprotov5.NewDynamicValue(nsVal.Type(), nsVal)
+	if err != nil {
+		resp.Diagnostics = append(resp.Diagnostics, &tfprotov5.Diagnostic{
+			Severity: tfprotov5.DiagnosticSeverityError,
+			Summary:  "Failed to construct dynamic value for imported state",
+			Detail:   err.Error(),
+		})
+		return resp, nil
+	}
+	resp.ImportedResources = append(resp.ImportedResources, &tfprotov5.ImportedResource{
+		TypeName: req.TypeName,
+		State:    &impState,
+	})
+	return resp, nil
+}
+
+// parseImportID processes the resource ID string passed by the user to the "terraform import" command
+// and extracts the values for GVK, name and (optionally) namespace of the target resource as required
+// during the import process.
+//
+// The expected format for the import resource ID is:
+//
+// "<apiGroup/><apiVersion>#<Kind>#<namespace>#<name>"
```

Comment:
Not sure about the `#` delimiter. We could just use the API path verbatim, then you don't really need to worry about slashes in the api version. 

`v1/namespaces/my-namespace/pods/my-pod`



As an alternative we could use key value pairs to be explicit like:

`apiVersion=v1,kind=Namespace,name=kube-system`

Reply:
the `#` also forces the whole thing to be quoted on the shell since it's the comment delimiter – you get a confusing error if you dont quote it.